.build-tools/cmd/zz-e2e-perf.go

```diff
@@ -417,25 +423,98 @@ func (c *cmdE2EPerf) getHashDir() (string, error) {
 			return nil
 		}
 
-		// Compute the sha256 of the file
-		checksum, err := checksumFile(path)
+		// Add the hash of the file
+		checksum, err := hashEntryForFile(path, relPath)
 		if err != nil {
 			return err
 		}
 
-		// Convert all slashes to / so the hash is the same on Windows and Linux
-		relPath = filepath.ToSlash(relPath)
-
-		files = append(files, relPath+" "+checksum)
+		files = append(files, checksum)
 		return nil
 	})
+
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+func hashEntryForFile(path string, relPath string) (string, error) {
+	// Compute the sha256 of the file
+	checksum, err := checksumFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	// Convert all slashes to / so the hash is the same on Windows and Linux
+	relPath = filepath.ToSlash(relPath)
+
+	return relPath + " " + checksum, nil
+}
+
+// Returns the checksum of the files in the directory
+func (c *cmdE2EPerf) getHashDir() (string, error) {
+	basePath := filepath.Join(c.getAppDir(), c.flags.Name)
+	_, err := os.Stat(basePath)
+	if err != nil {
+		fmt.Printf("could not find app %s\n", basePath)
+		return "", err
+	}
+
+	// Load the files to exclude
+	ignores := c.getIgnores()
+
+	// Load additional paths to include
+	includes := c.getIncludes()
+
+	// Compute the hash of the app's files
+	files, err := hashFilesInDir(basePath, ignores)
 	if err != nil {
 		return "", err
 	}
 	if len(files) == 0 {
 		return "", fmt.Errorf("no file found in the folder")
 	}
 
+	// Include other files or folders as needed
+	if len(includes) > 0 {
+		for _, pattern := range includes {
+			if !filepath.IsAbs(pattern) {
+				pattern = filepath.Join(basePath, pattern)
+			}
+			matches, err := filepath.Glob(pattern)
+			if err != nil || len(matches) == 0 {
+				continue
+			}
+			for _, match := range matches {
+				if match == "" {
+					continue
+				}
+				info, err := os.Stat(match)
+				if err != nil {
+					continue
+				}
+				if info.IsDir() {
+					addFiles, err := hashFilesInDir(match, nil)
```

Comment:
should we check for ignored files here?

I think ignores should have precedence over includes, WDYT?

Reply:
I did spend some thoughts on these.

My thought was that .gitignore files are usually specific for the folder they live in, and the includes are in other folders, so it felt "wrong" to apply the same .gitignores. Also, if the included paths are specific enough (you can see that I've tried to be specific, such as also adding `*.proto` or `*.go` at the end), then ignores shouldn't matter.

WDYT?

Reply:
Looks good, unless we have a usecase for including an entire folder and ignore a single file there, so the way to go in this scenario is a bit verbose than including dir/** and ignoring dir/onlythisfile

if we do, we can change it later

Reply:
can you add a comment on it explaining the rational to not use ignores?

Reply:
Done